core/libs/commonwealth/commonwealth/mavlink_comm/typedefs.py

```diff
@@ -66,6 +66,50 @@ class MavlinkVehicleType(str, Enum):
     MAV_TYPE_GPS = "Gps"
     MAV_TYPE_WINCH = "Winch"
 
+    def mavlink_firmware_type(self) -> str:
+        plane_options = [
+            MavlinkVehicleType.MAV_TYPE_FIXED_WING,
+            MavlinkVehicleType.MAV_TYPE_VTOL_DUOROTOR,
+            MavlinkVehicleType.MAV_TYPE_VTOL_QUADROTOR,
+            MavlinkVehicleType.MAV_TYPE_VTOL_TILTROTOR,
+            MavlinkVehicleType.MAV_TYPE_VTOL_FIXEDROTOR,
+            MavlinkVehicleType.MAV_TYPE_VTOL_TAILSITTER,
+            MavlinkVehicleType.MAV_TYPE_VTOL_RESERVED4,
+            MavlinkVehicleType.MAV_TYPE_VTOL_RESERVED5,
+        ]
+
+        copter_options = [
+            MavlinkVehicleType.MAV_TYPE_QUADROTOR,
+            MavlinkVehicleType.MAV_TYPE_COAXIAL,
+            MavlinkVehicleType.MAV_TYPE_HELICOPTER,
+            MavlinkVehicleType.MAV_TYPE_HEXAROTOR,
+            MavlinkVehicleType.MAV_TYPE_OCTOROTOR,
+            MavlinkVehicleType.MAV_TYPE_TRICOPTER,
+        ]
+
+        rover_options = [
+            MavlinkVehicleType.MAV_TYPE_GROUND_ROVER,
+            MavlinkVehicleType.MAV_TYPE_SURFACE_BOAT,
+        ]
+
+        sub_options = [
+            MavlinkVehicleType.MAV_TYPE_SUBMARINE,
+        ]
```

Comment:
I would personally prefer this class to be implemented the other way around (as a [union](https://gist.github.com/plammens/ab1a2f236b5c6d748f193eb12eefa6dd)/superset of the sub-groups), but assuming that's not reasonable at the moment it's at least worth making use of the object passed in to the function call rather than globally using the class name :-) 

```suggestion
        plane_options = [
            self.MAV_TYPE_FIXED_WING,
            self.MAV_TYPE_VTOL_DUOROTOR,
            self.MAV_TYPE_VTOL_QUADROTOR,
            self.MAV_TYPE_VTOL_TILTROTOR,
            self.MAV_TYPE_VTOL_FIXEDROTOR,
            self.MAV_TYPE_VTOL_TAILSITTER,
            self.MAV_TYPE_VTOL_RESERVED4,
            self.MAV_TYPE_VTOL_RESERVED5,
        ]

        copter_options = [
            self.MAV_TYPE_QUADROTOR,
            self.MAV_TYPE_COAXIAL,
            self.MAV_TYPE_HELICOPTER,
            self.MAV_TYPE_HEXAROTOR,
            self.MAV_TYPE_OCTOROTOR,
            self.MAV_TYPE_TRICOPTER,
        ]

        rover_options = [
            self.MAV_TYPE_GROUND_ROVER,
            self.MAV_TYPE_SURFACE_BOAT,
        ]

        sub_options = [
            self.MAV_TYPE_SUBMARINE,
        ]
```

Reply:
Code changed, but can you provide an example how it would like as union/superset ?

Reply:
I was thinking something like:

```python
from unionenum import enum_union, enum

def superset_of(*base_enums: enum.EnumMeta):
    def decorator(extension_enum: enum.EnumMeta):
        return enum_union(*base_enums, extension_enum)
    return decorator


class MAVLinkBaseType(str, Enum):
    @property
    def firmware_type(self) -> str:
        return f"Ardu{self.__class__.__name__[7:-4]}"


class MAVLinkSubType(MAVLinkBaseType):
    MAV_TYPE_SUBMARINE = "Submarine"


class MAVLinkRoverType(MAVLinkBaseType):
    MAV_TYPE_GROUND_ROVER = "Ground Rover"
    MAV_TYPE_SURFACE_BOAT = "Surface Boat"


... # other vehicle types


@superset_of(
    MAVLinkSubType,
    MAVLinkRoverType,
    ...,
)
class MAVLinkVehicleType(MAVLinkBaseType):
    MAV_TYPE_GENERIC = "Generic"
    ... # other miscellaneous types

    @property
    def firmware_type(self) -> str:
        return "Unknown"
```